contracts/finality/src/exec/finality.rs

```diff
@@ -63,7 +77,64 @@ pub fn handle_public_randomness_commit(
     Ok(Response::new().add_event(event))
 }
 
-// Copied from contracts/btc-staking/src/finality.rs
+/// Validates all parameters for public randomness commitment
+fn validate_pub_rand_commit(
+    env: &Env,
+    fp_btc_pk_hex: &str,
+    start_height: u64,
+    num_pub_rand: u64,
+    commitment: &[u8],
+    signature: &[u8],
+    min_pub_rand: u64,
+) -> Result<(), ContractError> {
+    // Check if FP BTC PubKey is empty
+    if fp_btc_pk_hex.is_empty() {
+        return Err(ContractError::EmptyFpBtcPubKey);
+    }
+
+    // Check if commitment is exactly 32 bytes
+    if commitment.len() != EXPECTED_COMMITMENT_LENGTH_BYTES {
+        return Err(ContractError::InvalidCommitmentLength {
+            expected: EXPECTED_COMMITMENT_LENGTH_BYTES,
+            actual: commitment.len(),
+        });
+    }
```

Comment:
https://github.com/babylonlabs-io/rollup-bsn-contracts/pull/58 will be subsumed by this PR cc @KonradStaniec?

Reply:
yep :) will close the old one